### PR TITLE
Put Statup Files in S3

### LINKED ISSUE #18: use pre-signed s3 for input files
Use s3 to store the input files, and presign the url before putting in transfer_input_files.  This allows us to not spool them to the schedd (creating N copies), and should greatly reduce disk usage.

From slack:
> this may be useful for you: https://github.com/WIPACrepo/iceprod/blob/master/iceprod/s3.py
just a wrapper around boto3 to make things async, and hide strange syntax.  not sure if there is a better library now

### PR DESCRIPTION
**Pull request by ric-evans**
closes #18

### CODE REVIEW

File: clientmanager/starter.py [hunk 1]
```diff
@@ -2,19 +2,59 @@
...
+    with open(filepath, "rb") as f:
+        response = requests.post(
+            upload_details["url"],
+            data=upload_details["fields"],
+            files={"file": (filepath.name, f)},  # maps filename to obj
+        )
+    print(f"Upload response: {response.status_code}")
+    print(str(response.content))
+
+    return S3File(upload_details["url"], filepath.name)
```
Comment (dsschult) [inline clientmanager/starter.py:56]: One note: you've used the `presigned_post` url here, which is correct for uploading.  But you need the presigned get url for downloading in the clients.  You probably want to generate that here and add it to S3File.  When generating the get url, you probably also want to set the expiration time (the default is 1 hour).

File: clientmanager/starter.py [hunk 2]
```diff
@@ -2,18 +2,60 @@
...
+def s3ify(filepath: Path) -> S3File:
+    """Put the file in s3 and return info about it."""
+    s3_client = boto3.client(
+        "s3",
+        "us-east-1",
+        endpoint_url=ENV.EWMS_TMS_S3_URL,
+        aws_access_key_id=ENV.EWMS_TMS_S3_ACCESS_KEY,
+        aws_secret_access_key=ENV.EWMS_TMS_S3_SECRET_KEY,
+    )
+    bucket = "clientmanager"
```
Comment (dsschult) [inline clientmanager/starter.py:38]: clientmanager

Comment (ric-evans) [reply clientmanager/starter.py]: got it :heavy_check_mark:

File: clientmanager/utils.py [hunk 3]
```diff
@@ -1,30 +1,99 @@
...
+        )
+    LOGGER.info(f"Upload response: {response.status_code}")
+    LOGGER.info(str(response.content))
 
-    return skydriver_rc, scan_id
+    # get GET url
+    get_url = s3_client.generate_presigned_url(
+        "get_object",
+        Params={"Bucket": bucket, "Key": key},
+        ExpiresIn=3600,  # 60 mins
```
Comment (dsschult) [inline clientmanager/utils.py:97]: We might want a longer expiration, if a scan takes more than 1 hour.  Maybe a day?

**Review by dsschult — APPROVED** [review]
Things generally look good.  It would be nice to test functions like `s3ify()` in CI, but not required.

File: clientmanager/utils.py (hunk 3 above)
Comment (ric-evans) [reply clientmanager/utils.py:97]: sure, I'll add an env var for that

**Pull request merged**